File: jetpack/src/main/java/com/example/jetpack/JetpackMainActivity.java
package com.example.jetpack;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.example.jetpack.lifecycle.LifecycleActivity;
import com.example.jetpack.livedata.ui.LiveDataMainActivity;
import com.example.jetpack.livedata.ui.LiveDataMapActivity;
import com.example.jetpack.navigation.NavigationMainActivity;
import com.example.jetpack.viewModel.ViewModelMainActivity;

public class JetpackMainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_jetpack_main);
    }

    public void onStartLifeCycle(View view) {
        startActivity(new Intent(JetpackMainActivity.this, LifecycleActivity.class));
    }

    public void onStartLiveData(View view) {
        startActivity(new Intent(JetpackMainActivity.this, LiveDataMainActivity.class));
    }

    public void onStartNavigation(View view) {
        startActivity(new Intent(JetpackMainActivity.this, NavigationMainActivity.class));
    }

    public void onStartLiveData2(View view) {
        startActivity(new Intent(JetpackMainActivity.this, LiveDataMapActivity.class));

    }

    public void onStartViewModel(View view) {
        startActivity(new Intent(JetpackMainActivity.this, ViewModelMainActivity.class));
    }
}

File: lib_kotlin/src/main/java/com/example/kotlin/_13kotlin/JetpackMainActivity.kt
package com.example.kotlin._13kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.kotlin.R
import com.example.kotlin._2kotlin.DevKotlin
import com.example.kotlin.startActivity
import kotlinx.android.synthetic.main.kotlin_13_jetpack_main_activity.*

@DevKotlin("开启 Jetpack 之旅")
class JetpackMainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.kotlin_13_jetpack_main_activity)

        viewModelBtn.setOnClickListener {
           startActivity<ViewModelActivity>(this)
        }

        lifecycleBtn.setOnClickListener {
            startActivity<LifecycleMainActivity>(this)
        }
        liveDataBtn.setOnClickListener {
            startActivity<LiveDataMainActivity>(this)
        }
        liveDataBtn2.setOnClickListener {
            startActivity<LiveDataMainActivity2>(this)
        }
        liveDataBtn3.setOnClickListener {
            startActivity<LiveDataTransformationActivity>(this)
        }
        liveDataBtn4.setOnClickListener {
            startActivity<LiveDataTransformationActivity2>(this)
        }
    }
}